packages/playwright-core/src/server/fetch.ts

```diff
@@ -312,8 +312,11 @@ export abstract class APIRequestContext extends SdkObject {
 
       let securityDetails: har.SecurityDetails | undefined;
 
+      const cleanup: (() => void)[] = [];
```

Comment:
looks like we want `eventsHelper.addEventListener` instead.

Reply:
that's a nifty helper! done in https://github.com/microsoft/playwright/pull/32956/commits/a483cab690f75424533739c97e89fc19cd251d5a